src/golang/lib/engine/aq_engine.go

```diff
@@ -742,13 +747,29 @@ func (eng *aqEngine) execute(
 	}
 
 	// Wait a little bit for all active operators to finish before exiting on failure.
-	defer waitForInProgressOperators(ctx, inProgressOps, timeConfig.OperatorPollInterval, timeConfig.CleanupTimeout)
+	defer func() {
+		waitForInProgressOperators(ctx, inProgressOps, timeConfig.OperatorPollInterval, timeConfig.CleanupTimeout)
+		if err != nil {
+			notificationLevel = mdl_shared.ErrorNotificationLevel
+			notificationCtxMsg = err.Error()
```

Comment:
Instead of an explicit `hasNotification` argument, I'd prefer if we defined a struct containing both the level and msg, and set some variable as a pointer to that struct. If that pointer is NULL, then there is no notification. That way it's all packaged behind one variable.